Format: add a guide (#1203)
SEKOIA-IO/documentation: _shared_content/develop/guides/formats/smartdescriptions.md

# How to write smart descriptions

> An event can be hard to read as a raw data. To make the visualization and the understand of events easier, smart-descriptions help to display important information to the user.

Reviewer: ```suggestion
An event can be hard to read as a raw data. To make the visualization and the understanding of events easier, smart-descriptions help displaying important information to the user.
```

## Organization

Smart-descriptions are defined in the directory `_meta/smart-descriptions.json` in the format. The document is written in [JSON](https://json.org) dialect.

The root element of the document is a JSON-array. This array contains a set of smart-descriptions.

## Smart-description

A smart description consists of:

- The value (mandatory):

	A sentence that will display information from the [structured event](structured_event.md).

	This sentence contains placeholders (delimited with curly-braces). Each placeholder encompasses a path (dot-notation); the path indicates the location of the value in the structured event that is used to replace the placeholder in the description.

	e.g:
	With a structured event containing:

	```json
	{
	    "source": {
	        "ip": "1.2.3.4"
	    }
	}
	```

	The description `Connection from {source.ip}` will result in `Connection from 1.2.3.4`.

- A list of conditions (optional):

	These conditions define when the description should be displayed. For a same event, the description, with the largest number of matching conditions, will succeed.

> 	A condition is compound of a path (dot-notation) to the field to test, and the expected value of the field (equal operation).

Reviewer: ```suggestion
	A condition is a compound of a path (dot-notation) to the field to test, and the expected value of the field (equal operation).
```

	e.g:
	With a structured event containing:

	```json
	{
	    "event": {
	      "kind": "alert"
	    }
	}
	```

	The condition

	```json
	{
	    "field": "event.kind",
	    "value": "alert"
	}
	```

	will match

	while the condition

	```json
	{
	    "field": "event.kind",
	    "value": "event"
	}
	```

	will not match

	To test the existence of a field in the structured event, only define the field in the condition.

	e.g:
	The condition

	```json
	{
	    "field": "event.kind"
	}
	```

	will test the existence of `event.kind`

- A list of relationships (optional): The relationships help to connect information from the structured event.

> 	A relationship is compound of:

Reviewer: ```suggestion
	A relationship is a compound of:
```
